client/credentials_test.go

```diff
@@ -0,0 +1,61 @@
+package client
+
+import (
+	"crypto/sha1"
+	"hash"
+	"net/http"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/assert"
+)
+
+func TestCredentials(t *testing.T) {
+
+	assert := assert.New(t)
+
+	credentials := &Credentials{
+		ClientID:    "tester",
+		AccessToken: "no-secret",
+	}
+
+	type CredentialsTests struct {
+		method string
+		url    string
+		host   string
+		port   int
+		hdr    string
+		now    int64
+		perr   error
+		verr   error
+		key    string
+		hash   hash.Hash
+		reply  bool
+		nonce  string
+		ext    string
+	}
+
+	test_credentials := &CredentialsTests{
```

Comment:
Just a small nit: [golint](https://github.com/golang/lint) is complaining about using snake case variable name. Could you please rename it to `testCredentials`?